runtime/gc_base/JavaObjectAllocationModel.hpp

```diff
@@ -77,7 +77,7 @@ class MM_JavaObjectAllocationModel : public MM_AllocateInitialization
 		if (NULL != objectPtr) {
 			/* Initialize class pointer in object header -- preserve flags set by base class */
 			MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(env);
-			extensions->objectModel.setObjectClass(objectPtr, _class);
+			extensions->objectModel.setObjectClass(objectPtr, getClass());
```

Comment:
Should the protected `_class` field become private to prevent subclass's from reaching for it directly?  And likely commented to indicate the issue.

Reply:
Making the field private means modifying a lot of code. Any users of the field before a GC has occurred really don't need to get the current class (the OM is used in the faster path NoGC code as well).

Reply:
Tempted to add a new accessor getCurrentClass() for people who are aware that the class may change during a GC, and leave getClass() as it is (there are actually no callers - all access is by direct field fetch).

Reply:
@dmitripivkine I'm OK with taking this change as is.  Do you want a followup issue to make the `_class` field private?

Reply:
I will not make the accessor change now - you can decide if you want it when changing the field visibility.